src/pages/PrintCards.tsx

```diff
@@ -1,73 +1,111 @@
-            </div>
-            <div
-              className="PrintingCards"
-              style={
-                allDimensions[index].type === "Practice"
-                  ? { backgroundColor: PracticeBGColour }
-                  : { backgroundColor: BeliefBGColour }
-              }
-            >
-              <span className="Print-Card-Text-TopLeft">{index + 1}</span>
-              <span className="Print-Card-Text">
-                {index % 2
-                  ? allDimensions[index].leftCard.statement
-                  : allDimensions[index].rightCard.statement}
-              </span>
-            </div>
-            <div className="PrintingCards">
-              <Dimension dimensionValue={index} SecondStatement={true} />
-            </div>
-          </>
-        ))}
-      </div>
-    );
+    const allDimensions =
+      this.props.passAllDimensions && this.props.passAllDimensions;
+    if (allDimensions) {
+      return (
+        <div>
+          {allDimensions.map((currElement: any, index: number) => (
+            <>
+              <div
+                className="PrintingCards"
+                style={
+                  allDimensions[index].type === "Practice"
+                    ? { backgroundColor: PracticeBGColour }
+                    : { backgroundColor: BeliefBGColour }
+                }
+              >
+                <span className="Print-Card-Text-TopLeft">{index + 1}</span>
+                <span className="Print-Card-Text">
+                  {index % 2
+                    ? allDimensions[index].rightCard.statement
+                    : allDimensions[index].leftCard.statement}
+                </span>
+              </div>
+              <div className="PrintingCards">
+                <Dimension
+                  dimensionValue={index}
+                  secondStatement={false}
+                  allDimensions={allDimensions[index]}
+                />
+              </div>
+              <div
+                className="PrintingCards"
+                style={
+                  allDimensions && allDimensions[index].type === "Practice"
+                    ? { backgroundColor: PracticeBGColour }
+                    : { backgroundColor: BeliefBGColour }
+                }
+              >
+                <span className="Print-Card-Text-TopLeft">{index + 1}</span>
+                <span className="Print-Card-Text">
+                  {index % 2
+                    ? allDimensions[index].leftCard.statement
+                    : allDimensions[index].rightCard.statement}
+                </span>
+              </div>
+              <div className="PrintingCards">
+                <Dimension
+                  dimensionValue={index}
+                  secondStatement={true}
+                  allDimensions={allDimensions[index]}
+                />
+              </div>
+            </>
+          ))}
+        </div>
+      );
+    } else {
+      return (
+        <div className="DisplayCards">
```

Comment:
This should use the classname for the Print component not DisplayCards. Furthermore, the loading symbol is not centred on the page during loading. This needs to be fixed.

Reply:
Yep forgot to change the code when I copied it over